x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -329,79 +346,100 @@ protected DataFrameTransformConfig getConfig() {
         }
 
         @Override
-        protected Map<String, String> getFieldMappings() {
-            return fieldMappings;
-        }
+        protected void onStart(long now, ActionListener<Void> listener) {
+
+            // NOTE: This is skipped if this is not the first run of the indexer
+            final DataFrameIndexerTransformStats stats = this.getStats();
+            ActionListener<SearchResponse> getTotalCountListener = ActionListener.wrap(
+                searchResponse -> {
+                    stats.setCurrentRunTotalDocumentsToProcess(searchResponse.getHits().getTotalHits().value);
+                    stats.resetCurrentRunDocsProcessed();
+                    super.onStart(now, listener);
+                },
+                e -> {
+                    stats.setCurrentRunTotalDocumentsToProcess(0L);
```

Comment:
Maybe leave the docs to process as null to distinguish between this failure and the case where there are no docs